Added calculation implementation for simple length objects

njlawton opened the pull request:
Implemented calculations between simple lengths and produced tests to check there functionality.

Started outline of position-value.js

Commit:
Created simple test case for multiplication of a simple length

Commit:
Implemented division of simple lengths
Implemented and porvided basic tests for the division method in the simple-length class

Commit:
Added tests for division of simple lengths

wilddamon commented:
Please give the PR a more descriptive name, and make sure to add the chai and mocha files to your .gitignore.

Commit:
Removed node_modules folder from git repo
Removed node_modules folder and updated the git ignore file to keep them untracked

njlawton commented:
Changed name of pull request and removed node_modules folder

wilddamon commented on src/length-value.js:
```diff
@@ -44,19 +44,31 @@
+    if(this instanceof SimpleLength && addedLength instanceof SimpleLength){
+      if(this.type === addedLength.type){
+        return this.addSimpleLengths(addedLength);
+      }
```
You should collapse these two if statements into one, or add an else case to the inner one. At the moment it's possible to return undefined. Likewise below in subtract.

wilddamon commented on src/position-value.js:
```diff
@@ -0,0 +1,32 @@
+  /**
+   * PositionValue(xPos, yPos)
+   */
+  function PositionValue(xPos, yPos) {
```
Consider moving this into a separate pull request - it's not related to adding calculation functionality to SimpleLength.

wilddamon commented on src/simple-length.js:
```diff
@@ -38,7 +38,24 @@
+  };
+
+  SimpleLength.prototype.addSimpleLengths = function(addedLength){
+    return new SimpleLength((this.value + addedLength.value), this.type);
```
Please add some validation/sanity checks that the units are the same, or consider moving logic for (eventually) creating CalcLengths into this function.

Commit:
Improved logic in methods in length_value.js
Collapsed two if statements into one in both the addition and subtraction methods

Commit:
Add CalcLength.cssString

 - Creates the cssString attribute for calcLength
   in the constructor.
 - Create a function in LengthValue to return the cssString
   representation of each LengthValue.LengthType
   [Note: uses switch/case block for extensibility]
   (LengthValue.cssStringTypeRepresentation)
 - Add appropriate cssString test cases
   (single, negative, multi, percent)

Commit:
Removed position-value.js from branch
This file is not relevant to current branch and was hence removed

Commit:
Added sanity checks to simple_length methods
Added object type and length type checking in the addition and subtraction helper methods in simple_length.js

Commit:
updated some function names
Due to renaming in simple-length.js code calling thease functions were changed in length-value.js

njlawton commented:
fixed issues stated above

wilddamon commented on src/simple-length.js:
```diff
@@ -38,7 +38,40 @@
+      } else {
+        throw new TypeError('SimpleLength units are not the same');
+      }
+    } 
```
The usual ordering for this is

if (error condition) {
// throw stuff
}

// do stuff

njlawton commented:
Fixed

wilddamon commented on src/length-value.js:
```diff
@@ -44,19 +44,27 @@
+  LengthValue.prototype.add = function(addedLength) {
+    if(this instanceof SimpleLength && addedLength instanceof SimpleLength && this.type === addedLength.type){
+      return this._addSimpleLengths(addedLength);
+    } else {
```
Don't need the else here since you are returning something in the if part.

wilddamon commented on src/simple-length.js:
```diff
@@ -38,7 +38,36 @@
+      if(!(this instanceof SimpleLength && addedLength instanceof SimpleLength)){
+        throw new TypeError('Objects not of type simple length');
+      }
+        throw new TypeError('SimpleLength units are not the same');
```
Please fix indentation, also it might be clearer to separate these into two non-nested statements

if (not both SimpleLengths) {
// throw something
}
if (different units) {
// throw something
}

// do stuff

Same below.

njlawton commented:
Solved git push problem

Commit:
Add createCssString closure in CalcLength

Seperate CssString creation from dictionary creation
in CalcLength constructor.
Uses closure createCssString(calcLen).

Commit:
Removed position value from pull request
Change relates to different part of system and hence is removed from the current pull request

Commit:
changed ordering of statements in simple-length.js
Changing statement ordering to check for errors first has now been used to remove braces

Commit:
Removed unnecessary else statement in length-value.js

nattridgell commented on test/js/simple-length.js:
```diff
@@ -25,4 +25,51 @@ suite('SimpleLength', function() {
+  test('Division of a simple length produses a new simple length object', function() {
+    var simpleLen = new SimpleLength(27, 'px');
+    var calcOutput = simpleLen.divide(3);
+    assert.strictEqual(simpleLen.type, 'px');
```
Was this meant to be checking calcOutput.type?

Commit:
Improved simple-length and length value to better adhere to conventions
added spaces around if statements and functions to better adhere to code conventions

Commit:
Changed simple-length.js tests to produse expected behaviour
Changed order of expected and actual statements to reflect how they are supposed to be layed out

Commit:
Fixed simple-length.js test cases
test cases were testing against the original length not the calculated one this has been fixed

Commit:
Changed if statements to simplify logic
Changed if statements in length-value.js and simple-value.js to simplify logic

Commit:
Merge branch 'master' of github.com:css-typed-om/typed-om into calculation

wilddamon commented on test/js/simple-length.js:
```diff
@@ -25,4 +25,51 @@ suite('SimpleLength', function() {
+    assert.strictEqual(calcOutput.value, 9);
+  });
+
+  test('Division of a simple length that contains decimals produses correct output value', function() {
```
s/produses/produces

wilddamon commented on test/js/simple-length.js:
```diff
@@ -25,4 +25,51 @@ suite('SimpleLength', function() {
+    assert.strictEqual(calcOutput.value, 9);
+  });
+
+  test('Multiplication of a simple length that contains decimals produses correct output value', function() {
```
s/produses/produces

wilddamon commented on test/js/simple-length.js:
```diff
@@ -25,4 +25,51 @@ suite('SimpleLength', function() {
     assert.strictEqual(valueFromNumber.value, 10);
   });
+
+  test('Multiplication of a simple length produses a new simple length object', function() {
```
s/produses/produces

Commit:
Fixed spelling mistakes in test/simple-length.js
